src/goto-symex/renaming_level.cpp

```diff
@@ -17,50 +17,53 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 #include "goto_symex_state.h"
 
-void symex_level0t::
-operator()(ssa_exprt &ssa_expr, const namespacet &ns, unsigned thread_nr) const
+ssa_exprt symex_level0t::
+operator()(ssa_exprt ssa_expr, const namespacet &ns, unsigned thread_nr) const
 {
   // already renamed?
   if(!ssa_expr.get_level_0().empty())
-    return;
+    return std::move(ssa_expr);
```

Comment:
@kroening @NathanJPhillips @smowton is the std::move actually necessary/desirable here (and in all the other cases where the argument is returned)? But maybe it doesn't matter much, because subsequent commits modify these lines.

Reply:
@tautschnig @romainbrenguier The `std::move` is not necessary; C++11 guarantees that the move constructor will be used without it, i.e., it's noise.

Reply:
There are three scenarios where `std::move` is needed:
1) The parameter is an rvalue reference
2) You want to move something whose lifetime isn't ending yet (but you are sure that you won't use it anymore)
3) You want to return something that's derived from the return type.
These aren't super intuitive.